infra/lint.sh

```diff
@@ -49,7 +49,7 @@ EOT
 }
 
 JAVA_FORMATTER="$(rlocation com_engflow_bazel_invocation_analyzer/third_party/google-java-format/google-java-format)"
-STARLARK_FORMATTER="$(rlocation com_engflow_bazel_invocation_analyzer/third_party/buildifier/buildifier.exe)"
+STARLARK_FORMATTER="$(rlocation com_engflow_bazel_invocation_analyzer/third_party/buildifier/buildifier)"
```

Comment:
I saw you apply this pattern internally. What's the advantage - why is this simpler and less prone to getting stale?

Reply:
> why is this simpler

Because it's no longer possible to mistype the `rlocation` argument. Such a typo would be found during runtime, i.e. it's a **run-time failure**.

By employing `rlocationpath`, it's Bazel that computes the argument that we need to pass to `rlocation`. And if we pass a wrong argument to `rlocationpath` then the build fails, so it's now a **build-time failure**.

> and less prone to getting stale

Because if we ever move one of these targets, let's say for example we move `//third_party/buildifier` to `@buildifier//:bin`, then we only need to update the BUILD file, not the shell script. With the old approach, we'd also need to update the shell script (= the string we pass to `rlocation`). See the run-time vs. build-time error check I described above.

Reply:
Thanks for clarifying, makes sense!